src/algokit_utils/applications/app_client.py

```diff
@@ -202,14 +253,19 @@ class AppClientCallParams:
     send_params: dict | None = None  # Parameters to control transaction sending
 
 
+ArgsT = TypeVar("ArgsT")
+MethodT = TypeVar("MethodT")
+OnCompleteT = TypeVar("OnCompleteT")
+
+
 @dataclass(kw_only=True, frozen=True)
-class AppClientMethodCallParams:
-    method: str
-    args: list[ABIValue | ABIStruct | AppMethodCallTransactionArgument | None] | None = None
+class BaseAppClientMethodCallParams(Generic[ArgsT, MethodT, OnCompleteT]):
```

Comment:
If these aren't exported, should be prefix with _? Are we using this convention or no real point?